src/com/civpvp/attrhider/AttrHider.java

```diff
@@ -33,129 +26,84 @@
-                    	}
-                    	ItemStack is = new ItemStack(i.getType(), 1 , (short) 1);
-                    	if (i.getItemMeta().getEnchants().values().size() != 0) {
-                    		is.addEnchantment(Enchantment.DURABILITY, 1);
-                    	}
-                    	if (color != null) {
-                    		LeatherArmorMeta lam = (LeatherArmorMeta) is.getItemMeta();
-                    		lam.setColor(color);
-                    		is.setItemMeta(lam);
-                    	}
-                        items.write(0, is);
-                    }
-                } catch (FieldAccessException exception) { //Should catch if the packet is the wrong type
-                    exception.printStackTrace();
-                }
-            }
-        });
-
-        //Strips potion duration length and sets it to 420 ticks so you can blaze it
-	    protocolManager.addPacketListener(new PacketAdapter(this, PacketType.Play.Server.ENTITY_EFFECT) {
-	    	@Override
-	    	public void onPacketSending(PacketEvent e) {
-	    		try {
-		    		PacketContainer p = e.getPacket();
-		    		if(e.getPlayer().getEntityId() != p.getIntegers().read(0)) { //Don't hide a player's own stuff from them
-		    			p.getIntegers().write(1, 420);
-		    		}
-	    		} catch (FieldAccessException exception) { 
-	    			exception.printStackTrace();
-	    		}
-	    	}
-	    });
-	    
-	    //Make reported health random
+	public void registerPacketAdapters() {
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EQUIPMENT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					ItemStack old = event.getPacket().getItemModifier().read(0);
+					if(old == null) return;
+					ItemStack clean = new ItemStack(old.getType());
+					if(old.hasItemMeta() && clean.hasItemMeta()) {
+						ItemMeta oldMeta = old.getItemMeta();
+						ItemMeta newMeta = clean.getItemMeta();
+						if(oldMeta instanceof LeatherArmorMeta) {
+							Color color = ((LeatherArmorMeta)oldMeta).getColor();
+							((LeatherArmorMeta)newMeta).setColor(color);
+						}
+						if(oldMeta.hasEnchants()) {
+							clean.addEnchantment(Enchantment.DURABILITY, 1);
+						}
+						if(oldMeta instanceof BannerMeta) {
+							DyeColor base = ((BannerMeta)oldMeta).getBaseColor();
+							List<Pattern> pattern = ((BannerMeta)oldMeta).getPatterns();
+							((BannerMeta)newMeta).setBaseColor(base);
+							((BannerMeta)newMeta).setPatterns(pattern);
+						}
+						clean.setItemMeta(newMeta);
+					}
+					event.getPacket().getItemModifier().write(0, clean);
+				}
+			});
+		
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EFFECT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					if(event.getPacket().getIntegers().read(0) != event.getPlayer().getEntityId()) {
```

Comment:
Why did you remove `PacketContainer p = e.getPacket();`  ? That was the correct pattern to use, when you access a field multiple times.